### PR TITLE
GH-54 - Search & Replace

### PR DESCRIPTION
**Pull request by auniverseaway**
* Creates new search tab in browse view
* Ability to search in current folder
* Ability to replace found search term

Resolves: GH-54

## How Has This Been Tested?

https://search--da-live--adobe.hlx.live/#/adobecom/bacom

## Screenshots (if appropriate):

Screenshot 2024-03-23 at 8 40 35 AM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [X] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [X] My code follows the code style of this project.
- [X] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [X] All new and existing tests passed.

## Release notes

* Creates new search tab in browse view
* Ability to search in current folder
* Ability to replace found search term

Resolves: <a href="https://github.com/adobe/da-live/issues/54">GH-54</a>

### CODE REVIEW

File: blocks/browse/da-browse/da-search.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    this.showReplace = !this.showReplace;
+  }
+
+  render() {
+    return html`
+      <form @submit=${this.handleSearch}>
+        <input type="text" placeholder="Enter search" name="term"/>
+        <input type="submit" value="Search" />
+      </form>
+      <p>${this.matches && this._total ? html`${this._action} ${this._matches} of ${this._total} ${this._time ? html`in ${this._time} seconds.` : nothing}` : nothing}</p>
```
Comment (chrischrischris) [inline blocks/browse/da-browse/da-search.js:146]: might be cleaner to assign this to a const instead of inlining

**Review by chrischrischris — DISMISSED** [review]

File: blocks/browse/da-browse/da-search.js (hunk 1 above)
Comment (auniverseaway) [reply blocks/browse/da-browse/da-search.js]: Good idea.

File: blocks/browse/da-browse/da-browse.css [hunk 2]
```diff
@@ -19,6 +19,62 @@ input, button {
...
+
+.da-browse-tab-header button {
+  position: relative;
+  display: block;
+  background: none;
+  border: none;
+  padding: 0 0 6px;
+  margin: 0;
+  box-sizing: border-box;
+  color: #8f8f8f;
```
Comment (bosschaert) [inline blocks/browse/da-browse/da-browse.css:37]: Would it be an idea to start using variables for colors so we can easily change the color scheme going forward if we want to?

**Review by bosschaert — APPROVED** [review]
LGTM

One thought I had is that since there is quite a lot of logic in the search/replace operation it would be good to add tests for it, probably a combination of unit tests and e2e tests. But we can do this separately.

**Comment by auniverseaway** [comment]
Agreed. We have a lot of room for improvement with tests. da-browse is due for a refactor (it's too big) and will write tests with that.

**Review by chrischrischris — APPROVED** [review]

**Pull request merged**